backend/apps/events/validators.py

```diff
@@ -0,0 +1,244 @@
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+from django.utils.timezone import make_aware
+from utils.helpers.validators import email_validation
+
+
+def time_validation(start_time_input, end_time_input, start_time=None, end_time=None):
+    """
+    Method for ensuring that any updates of time values are valid.
+    Takes in the new time inputs and potentially current values (if there are any)
+    """
+    # Make all date times aware (instead of naive)
+    if start_time_input is not None and start_time_input.tzinfo is None:
+        start_time_input = make_aware(start_time_input, is_dst=True)
+
+    if start_time is not None and start_time.tzinfo is None:
+        start_time = make_aware(start_time, is_dst=True)
+
+    if end_time_input is not None and end_time_input.tzinfo is None:
+        end_time_input = make_aware(end_time_input, is_dst=True)
+
+    if end_time is not None and end_time.tzinfo is None:
+        end_time = make_aware(end_time, is_dst=True)
+
+    # Check that input date times are after current time
+    if (start_time_input is not None and start_time_input < timezone.now()) or (
+        end_time_input is not None and end_time_input < timezone.now()
+    ):
+        raise ValidationError("Input datetimes are before current time")
+
+    # Check that the start time (either input or current value) is before the end time (either input or current value)
+    start_time_value = start_time
+    if start_time_input is not None:
+        start_time_value = start_time_input
+
+    end_time_value = end_time
+    if end_time_input is not None:
+        end_time_value = end_time_input
+
+    if start_time_value is not None and end_time_value is not None and start_time_value > end_time_value:
+        raise ValidationError(
+            "Invalid input: Start time must be before end time, and sign up open date must be before deadline"
+        )
+
+
+def title_and_desc_validation(title, description):
+    if title == "" or description == "":
+        raise ValidationError("Both title and description must be non-empty strings")
+
+
+def price_binding_sign_up_validation(attendable_data, attendable=None):
```

Comment:
I have to say that I believe this to be a lot easier to solve on a model-level by adding [conditional constraints](https://docs.djangoproject.com/en/4.0/ref/models/constraints/#checkconstraint) and then just catch the integrity error if it fails. Also applies to the datetime validations. By moving these the model-level and instead handling any errors that are raised, I believe we are less prone to bugs.

Reply:
Some of the logic here is also untested, which makes us prone to errors in the logic.